src/components/AddVariable/index.js

```diff
@@ -0,0 +1,167 @@
+              name="variableTarget"
+              className="variable-target"
+              type="text"
+              value={varTarget}
+              readOnly
+            />
+          </div>
+          <div className="var-modal">
+            <label htmlFor="varName">Variable Scope</label>
+            {
+              <ReactSelect
+                aria-label="Scope"
+                placeholder="Select a variable scope"
+                name="results"
+                value={scopeOptions.find((o) => o.value === inputScope)}
+                onChange={(selectedOption) =>
+                  setInputScope(selectedOption.value)
+                }
+                options={scopeOptions}
+                required
+              />
+            }
+          </div>
+          <div className="var-modal">
+            <label htmlFor="varName">Variable Name</label>
+            <input
+              id="varNameId"
+              name="varName"
+              className="addVarnameInput"
+              type="text"
+              value={inputName}
+              onChange={setInputName}
+            />
+          </div>
+          <div className="var-modal">
+            <label htmlFor="varName">Variable Value</label>
+            <input
+              id="varValueId"
+              name="varValue"
+              className="addVarValueInput"
+              type="text"
+              value={inputValue}
+              onChange={setInputValue}
+            />
+          </div>
+          <div className="form-input add-var-btn">
+            <a href="#" className="hover-state" onClick={closeModal}>
+              cancel
+            </a>
+            <Mutation mutation={addOrUpdateEnvVariableMutation}>
+              {(addOrUpdateEnvVariableByName, { called, error }) => {
+                let updateVar = varValues.map((varName) => {
+                  return varName.name;
+                });
+
+                updateVar = updateVar.includes(inputName);
+
+                if (error) {
+                  console.log(error)
+                  return <div>Unauthorized: You don't have permission to create
+                  this variable.</div>;
+                }
+
+                if (updateVar && called) {
+                  return <div>Updating variable</div>;
+                } else if (called) {
+                  return <div>Adding variable</div>;
+                }
+
+                const addOrUpdateEnvVariableHandler = () => {
+                  addOrUpdateEnvVariableByName({
+                    variables: {
+                      input: {
+                        name: inputName,
+                        value: inputValue,
+                        scope: inputScope,
+                        project: varProject,
+                        environment: varEnvironment,
+                      },
+                    },
+                  });
+                  setTimeout(() => {
+                    location.reload();
+                  }, "2000");
```

Comment:
no need for quotes, just `2000` would do. Also, should we do a hard reload or either:

- refetch data and update state, which auto updates the view
- or force a component rerender in a more 'hacky' way?